astropy/coordinates/builtin_frames/intermediate_rotation_transforms.py

```diff
@@ -108,27 +125,21 @@ def gcrs_to_tete(gcrs_coo, tete_frame):
 
 @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, TETE, GCRS)
 def tete_to_gcrs(tete_coo, gcrs_frame):
-    # compute the pn matrix, and then multiply by its transpose
-
-    jd1, jd2 = get_jd12(tete_coo.obstime, 'tt')
-    # Classical NPB matrix, IAU 2006/2000A
-    # (same as in builtin_frames.utils.get_cip).
-    rbpn = erfa.pnm06a(jd1, jd2)
+    # Compute the pn matrix, and then multiply by its transpose.
+    rbpn = erfa.pnm06a(*get_jd12(tete_coo.obstime, 'tt'))
     newrepr = tete_coo.cartesian.transform(matrix_transpose(rbpn))
-
-    gcrs = GCRS(newrepr, obstime=tete_coo.obstime, obsgeoloc=tete_coo.obsgeoloc,
-                obsgeovel=tete_coo.obsgeovel)
-
-    # now do any needed offsets (no-op if same obstime and 0 pos/vel)
+    # Transform to GCRS but still at the TETE location and obstime.
```

Comment:
Super-picky: the transformation has actually been done in the line above. So a different verb is needed - perhaps “Construct a GCRS coord” instead of “Transform to GCRS”.

Reply:
Indeed! Copy & pasted too quickly - will change it for CIRS too.

Reply:
Somewhat changed the wording overall, making it clearer for me at least...